pkg/networkservice/common/mechanisms/recvfd/server_test.go

```diff
@@ -0,0 +1,227 @@
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/chains/client"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/mechanisms/recvfd"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/mechanisms/sendfd"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/checks/checkcontext"
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcfdutils"
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcutils"
+	"github.com/networkservicemesh/sdk/pkg/tools/sandbox"
+)
+
+func createFile(fileName string, t *testing.T) (inodeURLStr string, fileClosedContext context.Context, cancelFunc func()) {
+	f, err := os.Create(fileName)
+	require.NoError(t, err, "Failed to create and open a file: %v", err)
+
+	err = f.Close()
+	require.NoError(t, err, "Failed to close file: %v", err)
+
+	fileClosedContext, cancelFunc = context.WithCancel(context.Background())
+
+	inodeURL, err := grpcfd.FilenameToURL(fileName)
+	require.NoError(t, err)
+
+	return inodeURL.String(), fileClosedContext, cancelFunc
+}
+
+func startServer(ctx context.Context, t *testing.T, testServerChain *networkservice.NetworkServiceServer, serveURL *url.URL) {
+	grpcServer := grpc.NewServer(grpc.Creds(grpcfd.TransportCredentials(insecure.NewCredentials())))
+	networkservice.RegisterNetworkServiceServer(grpcServer, *testServerChain)
+
+	errCh := grpcutils.ListenAndServe(ctx, serveURL, grpcServer)
+
+	require.Len(t, errCh, 0)
+}
+
+func createClient(ctx context.Context, u *url.URL) networkservice.NetworkServiceClient {
+	return client.NewClient(
+		ctx,
+		client.WithClientURL(sandbox.CloneURL(u)),
+		client.WithDialOptions(grpc.WithTransportCredentials(
+			grpcfd.TransportCredentials(insecure.NewCredentials())),
+		),
+		client.WithDialTimeout(time.Second),
+		client.WithoutRefresh(),
+		client.WithAdditionalFunctionality(sendfd.NewClient()))
+}
+
+func TestRecvfdClosesSingleFile(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+
+	dir := t.TempDir()
+
+	s, err := os.Create(path.Join(dir, "test.sock"))
+	require.NoError(t, err)
+
+	testFileName := path.Join(dir, t.Name()+".test")
+
+	inodeURLStr, fileClosedContext, cancelFunc := createFile(testFileName, t)
+
+	onFileClosedCallbacks := map[string]func(){
+		inodeURLStr: cancelFunc,
+	}
+
+	serveURL := &url.URL{Scheme: "unix", Path: s.Name()}
+
+	testChain := chain.NewNetworkServiceServer(
+		checkcontext.NewServer(t, func(t *testing.T, c context.Context) {
+			injectErr := grpcfdutils.InjectOnFileReceivedCallback(c, func(fileName string, file *os.File) {
+				runtime.SetFinalizer(file, func(file *os.File) {
+					onFileClosedCallback, ok := onFileClosedCallbacks[fileName]
+					if ok {
+						onFileClosedCallback()
+					}
+				})
+			})
+
+			require.NoError(t, injectErr)
+		}),
+		recvfd.NewServer())
+
+	startServer(ctx, t, &testChain, serveURL)
```

Comment:
This part is repeated in two tests.

Is it make sense to rework these tests on testify suite and move this setup logic into `SetupSuite` function?